### PR TITLE
fix crd generation

### PR DESCRIPTION
**Pull request by dhellmann**
Add a copy of operator-sdk v0.17.0 to the local tools directory and
modify it to support float fields. This gives us a version of the tool
that we can use in CI to verify future changes and add a new version
that drops the float field. At that point, we should remove this copy
of the tools and resuming using the published releases.

Addresses #434

### CODE REVIEW

**Comment by maelk** [comment]
In the CI, the BMH are in "inspecting" state, while if running the `openstack baremetal node list` they are both manageable.

**Comment by dhellmann** [comment]
> In the CI, the BMH are in "inspecting" state, while if running the `openstack baremetal node list` they are both manageable.

I can't imagine why the changes in this PR would cause that. Could that be related to the recent ironic image changes?

**Comment by maelk** [comment]
we did not see this problem anywhere else until now

**Comment by dhellmann** [comment]
OK, I'll look at it more closely then.

File: hack/shellcheck.sh [hunk 1]
```diff
@@ -7,7 +7,10 @@ CONTAINER_RUNTIME="${CONTAINER_RUNTIME:-podman}"
 
 if [ "${IS_CONTAINER}" != "false" ]; then
   TOP_DIR="${1:-.}"
-  find "${TOP_DIR}" -path ./vendor -prune -o -name '*.sh' -exec shellcheck -s bash {} \+
+  find "${TOP_DIR}" \
+       \( -path ./vendor -prune -o \
+       -path ./tools/operator-sdk -prune \) \
+       -o -name '*.sh' -exec shellcheck -s bash {} \+
```
Comment (zaneb) [inline hack/shellcheck.sh:13]: `find` is the worst unix command.

**Pull request merged**